configure

```diff
@@ -0,0 +1,79 @@
+#!/bin/sh
+
+##==============================================================================
+##
+## configuredir
+## rootrelative
+## root
+##     Determine the root directory of the distribution (not necessarily the
+##     same as the current directory, since configure can be run anywhere).
+##
+##==============================================================================
+
+configuredir=`pwd`
+rootrelative=`dirname $0`
+cd $rootrelative
+root=`pwd`
+cd $configuredir
+
+
+##==============================================================================
+##
+## Product, Version, and Bilddate information:
+##
+##==============================================================================
+
+product="OMIScriptProvider"
+fullproduct="OMI Provider for scripting languages for Linux"
+
+# MI_MAJOR and other items are dependent on a rational version number
+# As a result, if version file can't be found, consider it fatal and
+# don't generate a bogus one
+
+if [ -f ../../omiscriptprovider.version ]; then
+    . ../../omiscriptprovider.version
+fi
+
+if [ -n "$OMI_SCRIPT_PROVIDER_BUILDVERSION_MAJOR" -a -n "$OMI_SCRIPT_PROVIDER_BUILDVERSION_MINOR" -a -n "$OMI_SCRIPT_PROVIDER_BUILDVERSION_PATCH" -a -n "$OMI_SCRIPT_PROVIDER_BUILDVERSION_BUILDNR" ]; then
+    major=$OMI_SCRIPT_PROVIDER_BUILDVERSION_MAJOR
+    minor=$OMI_SCRIPT_PROVIDER_BUILDVERSION_MINOR
+    revision=$OMI_SCRIPT_PROVIDER_BUILDVERSION_PATCH
+    patch_level=$OMI_SCRIPT_PROVIDER_BUILDVERSION_BUILDNR
+else
+    echo "*** UNABLE TO LOCATE FILE omiscriptprovider.version; USING VERSION 1.1.1.1 ***" >& 2
```

Comment:
This sort of recovery is from other packages, that are sometimes built without proper version files (although that's no longer the case, I believe).

Why do you need this here? Why not just error out?

Reply:
Done.

Reply:
Actually that is needed for Travis CI which builds the omiscriptprovider without having omiscriptprovider.version file.

Reply:
If it’s needed for Travis CI, you have two ways of solving that:

1. Check out the super-project, then replace your repo with the one from the super project. You can see how SCXCore does this. This is a better way of doing this: That way, you have all dependencies, can actually build a kit (since you have PAL dependency), etc. This makes the Travis CI build much more like the system build (a good thing), or

2. Have the travis.yml file create one for purposes of the build. Not as good, but that will work too.

I think both of these options are better than putting support code in configure for this.

Let me know your thoughts ...

Reply:
I agree with the first option, PR updated. Thanks for the suggestion.